part/set.go

```diff
@@ -88,6 +77,12 @@ func (s Set[T]) All() SetIterator[T] {
 
 // Union combines the values in the two sets. Returns a new set.
 func (s Set[T]) Union(s2 Set[T]) Set[T] {
+	if s2.tree == nil {
+		return s
```

Comment:
Doesn't return a new set like the comment claims.

Reply:
Reworded.